### ISSUE TITLE
Batali resolution fails with a single cookbook

### ISSUE THREAD

**Issue by luckymike** [hw-labs/batali#4]
With 1 cookbook in the Batali file:
```
Batali.define do
  source 'https://supermarket.chef.io'
  cookbook 'chef-server-populator', path: '../cookbooks/chef-server-populator'
end
```
I get the following error: 
```
[Batali]: Loading sources... error!
[ERROR]: Reason - Unable to coerce given type `Bogo::Smash` to `Batali::BFile::Cookbook`!
ERROR: ArgumentError: Unable to coerce given type `Bogo::Smash` to `Batali::BFile::Cookbook`!
```
With a second cookbook defined:
```
Batali.define do
  source 'https://supermarket.chef.io'
  cookbook 'chef-server-populator', path: '../cookbooks/chef-server-populator'
  cookbook 'users'
end
```
It works: 
```
[Batali]: Loading sources... complete!
[Batali]: Performing single path resolution.
[Batali]: Resolving dependency constraints... complete!
[Batali]: Writing manifest... complete!
[Batali]: Number of solutions collected for defined requirements: 1
[Batali]: Ideal solution:
chef-server<3.0.1>
chef-server-ingredient<0.2.0>
chef-server-populator<0.4.1>
packagecloud<0.0.17>
users<1.8.0>
```

**Comment by chrisroberts** [comment]
Need to add checks on collapsible items. Some what related commit cleaning up the same thing: https://github.com/hw-labs/batali/commit/ef9693fa9f95622fbd9984b4b4c83f4bea71ee53

**Comment by chrisroberts** [comment]
Fixed in v0.1.6

**Comment by luckymike** [comment]
I'm still having failures with v0.1.8:
```
[Batali]: Loading sources... error!
[ERROR]: Reason - undefined method `last' for #<Bogo::Smash:0x00000101148d48>
ERROR: NoMethodError: undefined method `last' for #<Bogo::Smash:0x00000101148d48>

Batali.define do
  source 'https://supermarket.chef.io'
  cookbook 'chef-server-populator', git: 'https://github.com/hw-cookbooks/chef-server-populator.git', ref: 'feature/chef-12'
end
```

**Comment by chrisroberts** [comment]
Updated how the data is structured when reading in the file to force expected type. Fix is available in v0.1.10
